packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts

```diff
@@ -0,0 +1,127 @@
+import { ICheckboxStyles } from './Checkbox.Props';
+import {
+  ITheme,
+  mergeStyleSets,
+  getFocusStyle
+} from '../../Styling';
+import { memoizeFunction } from '../../Utilities';
+
+const MS_CHECKBOX_LABEL_SIZE = '20px';
+const MS_CHECKBOX_TRANSITION_DURATION = '200ms';
+const MS_CHECKBOX_TRANSITION_TIMING = 'cubic-bezier(.4, 0, .23, 1)';
+
+export const getStyles = memoizeFunction((
+  theme: ITheme,
+  customStyles?: ICheckboxStyles
+): ICheckboxStyles => {
+  const { semanticColors, palette } = theme;
+  const checkmarkFontColor = semanticColors.inputForegroundChecked;
+  const checkboxBorderColor = semanticColors.inputBorder;
+  const checkboxBorderHoveredColor = semanticColors.inputBorderHovered;
+  const checkboxBackgroundChecked = semanticColors.inputBackgroundChecked;
+  const checkboxBackgroundCheckedHovered = semanticColors.inputBackgroundCheckedHovered;
+  const checkboxBackgroundDisabled = semanticColors.disabledText;
+  const checkboxTextColorDisabled = semanticColors.disabledText;
+
+  const styles: ICheckboxStyles = {
+    root: [
+      getFocusStyle(theme),
+      {
+        padding: '0',
+        border: 'none',
+        background: 'none',
+        margin: '0',
+        outline: 'none',
+        display: 'block',
+        cursor: 'pointer',
+
+        // This is to hide the label's overflow because of the negative margin.
+        overflow: 'hidden'
+      }
+    ],
+    label: {
+      display: 'inline-flex',
+      alignItems: 'center',
+      cursor: 'pointer',
+
+      // The negative margin is in place to correct for the margin of the box
+      margin: '0 -4px',
+      position: 'relative',
+      userSelect: 'none'
+    },
+    labelReversed: {
+      flexDirection: 'row-reverse',
+      justifyContent: 'flex-end'
+    },
+    labelDisabled: {
+      cursor: 'default'
+    },
+    input: {
+      position: 'absolute',
+      opacity: '0'
+    },
+    box: {
+      display: 'flex',
+      margin: '0 4px',

```

Comment:
I do not know. But, if I have to make a guess, it would be to address RTL styles too.